web/app.go

```diff
@@ -70,6 +71,7 @@ type Dependencies struct {
 	settingsService      services.SettingsService
 	telemetryRegistry    *telemetry.TelemetryRegistry
 	telemetryPublisher   telemetry.Publisher
+	premiumDetection     services.PremiumDetectionService
```

Comment:
I would add the `Service` suffix here (and in general, in the whole file)l
- `NewPremiumDetectionService`
- `premiumService`

Reply:
changed.
I find the suffix a bit of redundant, buy I don't mind, it's just details. 
There are other critical things :smile: